### PR TITLE
Main Menu Overhaul

### LINKED ISSUE #18: Menu for Settings
Description:
We need to design and implement a settings menu that allows users to customize key game parameters. The menu will handle various user preferences, such as audio levels  and control configurations. The focus is on creating a modular, scalable interface that can manage user input and update the system's configuration in real time.

Steps to Reproduce:
Start by structuring the user interface (UI). This involves designing a hierarchy of components, where each component (e.g., buttons, sliders, dropdowns) will map to specific game parameters. The UI needs to be event-driven, meaning it should respond dynamically to user actions such as clicks or drags, triggering appropriate callbacks in the underlying system.

Each setting option (e.g., audio volume) should be tied to a data-binding mechanism that reflects changes made in the UI directly into the game state. For instance, a slider's value should update the audio system’s global volume variable, while dropdowns should invoke methods to adjust the graphics quality level.

The logic for handling changes should adhere to the Model-View-Controller (MVC) or Model-View-ViewModel (MVVM) pattern, separating the user interface from the underlying system logic. Each setting element acts as the "View," which displays and captures user input. The "Controller" or "ViewModel" processes these inputs, translating them into actionable changes within the system.

Implement a state persistence mechanism so that user settings are saved and loaded across game sessions. This will likely involve serialization of the user's preferences (e.g., JSON, PlayerPrefs) and loading the saved state upon game initialization.

Finally, ensure that the settings menu is accessible and modular. Implement an event-handling system so that actions like applying or reverting settings trigger state changes, updating the game in real time. The UI should be modular, allowing different sections (audio, graphics, controls) to be extended or modified without affecting other parts of the system.

Expected Outcome:
The system should manage user input effectively, applying game-wide changes in response to the user's configuration choices. All settings should persist between sessions, and the interface should remain decoupled from game logic for easy maintenance and scalability.

Acceptance Criteria:
The settings menu should be structured using a clean separation of UI and system logic.
User preferences should be captured, processed, and stored persistently.
Changes should be reflected in real-time with an event-driven update mechanism.
The system should be easily extendable, allowing new settings to be added without significant restructuring.

### LINKED ISSUE #112: Main Menu Overhaul
To complete this issue , a settings UI should be added when the user clicks on the settings button allowing them to toggle / change  specific settings. In addition, a tutorial UI should be added explaining instructions of how to play the game when the user clicks on the tutorial button.

### PR DESCRIPTION
**Pull request by Zahid2104**
This issue adds the setting UI within the main menu which can be accessed by the user by selecting the settings button. Within this UI users can change settings to thier personal prefereneces such as ability to change NPC Subtitle speed, Camera sensetivity, SFX sound and music sounds. The issue also adds a tutorial screen which can be selected when the user clicks on the tutorial button within the main menu providing them with information such as the aim of the game and instructions on how to play the game. In addition the issue also adds a new load game button which is yet to be implemented and a few design changes such as new button colours and borders and a background image 

Closes #112 
Closes #18

### CODE REVIEW

File: Assets/Scripts/Menu/TutorialMenuScript.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            //set button clicked methods
+            _backButton.clickable.clicked += BackToMenu;
+        }
+
+        private void BackToMenu()
+        {
+            TutorialMenu.gameObject.SetActive(false);
+            MainMenu.gameObject.SetActive(true);
+        }
+    }
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Menu/TutorialMenuScript.cs:29]: add summaries to these methods

File: Assets/Scripts/Menu/OptionMenuScript.cs [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UIElements;
+using AudioManager = Scripts.Audio.AudioManager;
+using GameSettings = Scripts.Game.GameSettings;
+
+namespace Scripts.MainMenu
+{
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Menu/OptionMenuScript.cs:11]: The File is named menu so the namespace should be `Scripts.Menu`

File: Assets/TutorialMenu.uss
Comment (Amal-Mathew204) [inline Assets/TutorialMenu.uss:1]: Should be place the USS files in a MenuStyles folder?

**Review by Amal-Mathew204 — CHANGES_REQUESTED** [review]

**Review by Aylinx13 — CHANGES_REQUESTED** [review]
1.  You're awarded microchips not tokens. 
2.  Mention controls for walking
3. Mention controls for camera
4. Mention controls for pause menu
5. Mention controls for accepting tasks (f key)
6. Mention the enter key for the start npc
7. Change the description of the m key to be for the active mission log
8. Mention you have to stop and look at the item to pick it up

**Comment by Amal-Mathew204** [comment]
Background Images for this Main Menu should be placed in an styles folder specific to the main menu

**Review by Amal-Mathew204 — APPROVED** [review]

**Review by Amal-Mathew204 — APPROVED** [review]

**Review by Aylinx13 — APPROVED** [review]

**Pull request merged**